### ISSUE TITLE
Metadane przy wyborze słownika

### ISSUE THREAD

**Issue by robertdrozd** [PMKrol/CC-SJP#4]
Gdy wybieramy słownik na PB, pojawia się pod etykietą "Inne", a nie "Polish". Może to kwestia jakichś metadanych, które nie zostały ustawione?

![scr0200_wybor_slownika](https://user-images.githubusercontent.com/126499902/221619187-d44daedc-382b-4535-a22a-3edf4c2fd744.png)

**Comment by madalinski** [comment]
Sprawdziłem chyba wszystkie możliwe kombinacje z tagami, ale jak dotąd bez powodzenia 🙃

**Comment by PMKrol** [comment]
Zobaczmy co na [mobileread](https://www.mobileread.com/forums/showthread.php?t=325203&page=14) nam powiedzą ;)

**Comment by madalinski** [comment]
Trzymam kciuki - sprawdzałem kilka ze słowników tam wrzucanych przy okazji analizy formatu nagłówków w xdxf... i w żadnym to nie działało 🫣 Chyba trzeba pójść głębiej, do źródła converter.exe
Edit: https://www.mobileread.com/forums/showpost.php?p=3923322&postcount=6

**Comment by PMKrol** [comment]
W międzyczasie: [to](https://www.mobileread.com/forums/showpost.php?p=3923458&postcount=11).

**Comment by PMKrol** [comment]
W kodzie convertera można znaleźć ciągi znaków:

> any key to continue --�k�full_name�b�description�ar�?xml�xdxf�i�**lang_from�lang_to**�unclosed xml tag�LANG_TO�LANG_FROM�internal error: *zpos != 0�&amp;�&lt;�&gt;�&quot;�cannot open input XDXF file�this file is not in XDXF format�1.2.3������� deflate 1.2.3 Copyright 1995-2005 Jean-loup Gailly ����

Jeszcze nie próbowałem tego dekompilować i analizować, bo to nie jest rzecz którą często robię ;)
...ale ewidentnie program tych słów szuka. Trzeba tylko dowiedzieć się gdzie... 
![image](https://user-images.githubusercontent.com/88658026/221818479-223fd975-db47-42f9-90f7-b6b3a0b3764b.png)

@madalinski , czy mógłbyś wypróbować to: `<xdxf lang_from="eng" lang_to="eng" format="visual">`
+ WIELKIMI LITERAMI w różnych miejscach (jako samodzielny tag i ... gdzieś?).

**Comment by madalinski** [comment]
![obraz](https://user-images.githubusercontent.com/2414491/221826241-dff64cea-a217-45d2-ab54-4e13658e0998.png)
```
<?xml version="1.0" encoding="UTF-8" ?>
<xdxf lang_from="eng" lang_to="eng" format="visual">
    <full_name>tag xdxf lang_from eng lang_to eng</full_name>
```
```
<?xml version="1.0" encoding="UTF-8" ?>
<xdxf LANG_FROM="eng" LANG_TO="eng" format="visual">
    <full_name>w tagu xdxf LANG_FROM eng LANG_TO eng</full_name>
```
```
<?xml version="1.0" encoding="UTF-8" ?>
<xdxf LANG_FROM="ENG" LANG_TO="ENG" format="visual">
    <full_name>w tagu xdxf LANG_FROM ENG LANG_TO ENG wielkie</full_name>
```
```
<?xml version="1.0" encoding="UTF-8" ?>
<xdxf lang_from="eng" lang_to="eng" format="visual">
    <full_name>tag xdxf lang_from eng lang_to eng</full_name>
```
```
<?xml version="1.0" encoding="UTF-8" ?>
<lang_from "eng">
<lang_to "eng">
<xdxf format="visual">
    <full_name>testowy - przed xdxf w tagach lang_from lang_to</full_name>
```
```
<?xml version="1.0" encoding="UTF-8" ?>
<xdxf format="visual">
<lang_from>eng</lang_from>
<lang_to>eng</lang_to>
    <full_name>testowy - za xdxf</full_name>
```
```
<?xml version="1.0" encoding="UTF-8" ?>
<xdxf format="visual">
<lang_from "eng">
<lang_to "eng">
    <full_name>testowy - za xdxf w tagach lang_from lang_to</full_name>
```
```
<?xml version="1.0" encoding="UTF-8" ?>
<xdxf>
<from xml:lang="en"/>
<to xml:lang="en"/>
    <full_name>xml lang za xdxf</full_name>
```

**Comment by madalinski** [comment]
Eureka!
![obraz](https://user-images.githubusercontent.com/2414491/221832430-304bba79-b08c-4aaa-a429-ffbef0498aed.png)
Tyle, że takie rozwiązanie odpada dla większości użytkowników.
W katalogach PB znalazłem plik bazy danych z rozpisanymi słownikami, i dodałem odpowiedni wpis 😅
![obraz](https://user-images.githubusercontent.com/2414491/221832798-e1b141e9-7b8d-421b-b810-49c4932edc04.png)

**Comment by PMKrol** [comment]
Grubo, ale niekoniecznie jest to zła opcja.
Zakładając, że większość użytkowników nie zmienia tej bazy, można by udostępniać słownik wraz z pbdicts _podstawowym_ z dopisanym słownikiem cc-sjp. Da się wpisać w file gwiazdkę? Żeby wszystkie SJP-PocketBook-*.dic rozpoznawał jako PL?

edit. udostępnij mi proszę, jeśli możesz, plik bazy.
edit2. O! i plik słownika, który działa natywnie z oznaczeniem języka. Może tam coś znajdziemy.

**Comment by PMKrol** [comment]
```
<?xml version="1.0" encoding="UTF-8" ?>
<xdxf format="visual">
<lang_from>eng</lang_from>
<lang_to>eng</lang_to>
    <full_name>testowy - za xdxf</full_name>
```

a to wielkimi literami?

**Comment by madalinski** [comment]
Wpisać się da, ale gwiazdka nic nie daje.

Próbowałem wielkimi, również bez "full_name" (skoro po wpisaniu do bazy nazwę "połyka"), ale nic z tego.

[pbdicts.zip](https://github.com/PMKrol/CC-SJP/files/10850190/pbdicts.zip)
![obraz](https://user-images.githubusercontent.com/2414491/221853680-cf507082-3bb0-4eec-ab4e-82cc57be5937.png)

**Comment by PMKrol** [comment]
A jakbyś skopiował plik ze słownikiem, który działa natywnie z inną nazwą - wpada do Other czy tam gdzie trzeba?

**Comment by madalinski** [comment]
Do natywnych słowników niestety nie mam dostępu - prawdopodobnie są na jakiejś chronionej partycji. 
Próbowałem je znaleźć i podejrzeć co mają w środku, i tak własnie trafiłem na tę bazę.

**Comment by PMKrol** [comment]
![image](https://user-images.githubusercontent.com/88658026/221857824-fda94963-6f1b-4ee4-91ec-67a52c35d9d1.png)
To gdzieś tu jest! Ale chyba jestem za wąski na taką głęboką analizę softu...

**Comment by PMKrol** [comment]
@madalinski, wysłałem Tobie maila ze słownikami wypakowanymi z firmware PocketBooka ^^

Użyłem [tego](https://github.com/Vanuan/pocketbook_free_swupdate) na [tym](https://www.pocketbook-int.com/ge/support/pocketbook-inkpad-3).

**Comment by madalinski** [comment]
Po wrzuceniu na PB najpierw wylądowały we właściwych kategoriach, więc zmieniłem nazwy plików... i wleciały do "other" 😅
Edit: Był jeden wyjątek - "English_German.dic" po zmianie nazwy na "angielsko-niemiecki.dic" nadal był w kategorii "English" ... ale wychodzi na to, że jedynie dlatego, że w nazwie miał "English-German" (które było kompatybilne z wpisem w bazie), gdy zmieniłem to (nazwę) w pliku *.dic to dołączył do reszty.

**Comment by PMKrol** [comment]
No, to chyba mamy case closed jako niemożliwe do rozwiązania. Skoro nawet oryginalne słowniki nie są rozpoznawane poprawnie, to tylko modyfikacja bazy danych.

Dzięki.
